src/cudadecoder/batched-threaded-nnet3-cuda-online-pipeline.cc

```diff
@@ -273,13 +300,26 @@ void BatchedThreadedNnet3CudaOnlinePipeline::DecodeBatch(
     const std::vector<int> &n_input_frames_valid,
     const std::vector<BaseFloat *> &d_ivectors,
     const std::vector<bool> &is_first_chunk,
-    const std::vector<bool> &is_last_chunk, std::vector<int> *channels) {
+    const std::vector<bool> &is_last_chunk, std::vector<int> *channels,
+    std::vector<const std::string *> *in_partial_hypotheses,
+    std::vector<bool> *in_end_points) {
   nvtxRangePushA("DecodeBatch");
   if (!channels) {
     channels = &channels_;
     ListIChannelsInBatch(corr_ids, channels);
   }
 
+  partial_hypotheses_ = in_partial_hypotheses;
+  end_points_ = in_end_points;
+  {
+    std::lock_guard<std::mutex> lk(map_callbacks_m_);
+    if (!best_path_callbacks_.empty()) {
+      // If best path callbacks are in use, always activate partial hyp and endp
```

Comment:
I do not understand why best path callback implies partial hypotheses and endpointing. In the binary sample both are gladly ignored in case the user did not ask for them. Are they inseparable?
https://github.com/kaldi-asr/kaldi/blob/38b80369ded5609b39a7dc8482f29c865e0cd3f7/src/cudadecoderbin/batched-wav-nnet3-cuda-online.cc#L162-L185

Reply:
Both are used as arguments for the best path callbacks, and once we have one having the other is pretty cheap anyway

Reply:
Got it, thanks!